packages/medusa/src/repositories/gift-card.ts

```diff
@@ -45,15 +51,19 @@ export class GiftCardRepository extends Repository<GiftCard> {
     const entitiesAndRelations = entitiesIdsWithRelations.concat(entities)
 
     const entitiesAndRelationsById = groupBy(entitiesAndRelations, "id")
-    return Object.values(entitiesAndRelationsById).map((v) => merge({}, ...v))
+    return [
+      Object.values(entitiesAndRelationsById).map((v) => merge({}, ...v)),
```

Comment:
suggestion: Would it be possible to use ` merge(this.create(), ...v)` wdyt?

Reply:
Not sure how this would work, we are not creating anything but just merging the results of multiple queries - let me know what you had in mind? :)

Reply:
no worries, the thing is that i just tested now to test my feelings and each object of that result is of type any, so we are kind of casting the result and I would have expect that with that we shouldn't have to do that but in fact it does not change anything. The any type comes from the result of the merge as expected.